src/app/_components/client/Avatar/Avatar.tsx

```diff
@@ -0,0 +1,64 @@
+"use client";
+
+import profileImage from "../../../../../public/assets/userProfile.svg";
+import { AVATAR_ICON, AVATAR_SIZE } from "./Avatar.constant";
+import { useChangeImage, useClickAvatar } from "./hooks";
+
+import Image from "next/image";
+
+export interface AvatarProps {
+  type?: "view" | "edit";
+  size?: "small" | "medium" | "large";
+  avatarImage: string | null;
+  userId?: string;
+  onChangeAvatar?: (file: File | null) => void;
+}
+
+const Avatar = ({
+  type = "view",
+  size = "small",
+  avatarImage = null,
+  userId,
+  onChangeAvatar,
+}: AvatarProps) => {
+  const { preview, inputRef, handleChangeImage } = useChangeImage({ avatarImage, onChangeAvatar });
+  const { handleClickAvatar } = useClickAvatar({ type, inputRef, userId, onChangeAvatar });
+  const avatarImageSize = AVATAR_SIZE[size];
+  const avatarIconSize = AVATAR_ICON[size];
+
+  return (
+    <div
+      className={`relative ${avatarImageSize} ${(userId || type === "edit") && "cursor-pointer"}`}
+      onClick={handleClickAvatar}
+    >
+      <div className="size-[100%] bg-gray-300 rounded-circle  overflow-hidden">
+        <Image
+          className="object-cover rounded-circle"
+          src={preview || profileImage}
+          alt="프로필 이미지 사진"
+          sizes="100%"
+          fill
+          priority
+        />
+      </div>
+
+      {type === "edit" && (
+        <>
+          <span
+            className={`absolute bottom-0 right-0 ${avatarIconSize} bg-[#F1F1F1] rounded-full`}
+          ></span>
+          <input
+            className="hidden"
+            ref={inputRef}
+            type="file"
+            name="profile"
+            id="profile"
+            onChange={handleChangeImage}
+          />
+        </>
+      )}
```

Comment:
[ P2 ]

해당 Avatar Component가 특히 `RSC`로 제작되면 좋을것 같다고 생각이 들었습니다!

현재 Edit Type때문에 RSC가 아닌 RCC로 제작되어지는것같은데 아래 처럼 Image 태그와 Input을 별도의 하나의 `RCC`로 제작하고 해당 Avatar 의 기본은 `RSC`로 가져가는건 어떨까요?

```suggestion
        {type === "view" && (
          <Image
            className="object-cover rounded-circle"
            src={preview || profileImage}
            alt="프로필 이미지 사진"
            sizes="100%"
            fill
            priority
          />
        )}
      </div>
      
      
// --------- 아래 부분을 하나의 Component로 분리! 하면 어떨까요?


      {type === "edit" && (
        <>
          <Image
            className="object-cover rounded-circle"
            src={preview || profileImage}
            alt="프로필 이미지 사진"
            sizes="100%"
            fill
            priority
          />
   
          <span
            className={`absolute bottom-0 right-0 ${avatarIconSize} bg-[#F1F1F1] rounded-full`}
          ></span>
          <input
            className="hidden"
            ref={inputRef}
            type="file"
            name="profile"
            id="profile"
            onChange={handleChangeImage}
          />
        </>
      )}
```

Reply:
저도 현재와 같이 `Avatar` 컴포넌트 내부에서 `type`을 받아 `view, edit`을 분리하는 로직이라면 각각을 컴포넌트로 분리하는 것이 좋을 것 같습니다 !!

수정하겠습니다 ! 😊

Reply:
코드를 수정하다가 문득 생각이 들었습니다 ..!

해당 컴포넌트는 `edit`이 아니어도 Avatar 자체에 대한 `click 이벤트`가 필요하기 때문에 client 컴포넌트로 가야 합니다. 이런 상황에서도 어떻게 분리하면 상위 부모 태그는 서버 컴포넌트로 만들 수 있겠죠..! 그렇지만 상위 태그 하나 존재하는 `Avatar`로 만든 느낌이 되는 건데.. 이렇게 해도 사실상 상위 뎁스를 제외하면 RCC 다보니 hydration이 일어나야 하위 컴포넌트가 동작하지 않나요?? 뎁스만 깊어지는 건 아닌지 라는 생각이 들었습니다.

또 하나는 지금과 다르게 Image 태그를 공용으로 사용하지 않을 거면 Avatar를 하나의 컴포넌트로 사용하는 것보다 분리해서 사용하는 측면이 더 좋지 않을까라는 생각도 들었네요 😂

Reply:
분리하는것도 좋은 생각인것 같네요 그리고 RCC의 코드 비용이 적게 들어간다면 RSC로 만들 필요는없을것같네요.

위 코드상태로는 저부분을 분리하면 RSC로 보여질 수 있을거라고 생각했는데 edit 기능외 일반 Avatar도 이벤트가 존재하면 RCC도 좋은 선택인것같아요!